### PR TITLE
Refactor the code of vineyardctl and vineyard operator

### LINKED ISSUE #1234: Refactor similar code in vineyardctl and vineyard operator to improve readability
Describe your problem
---------------------

As titled. We can use tools such as design patterns to achieve code reuse.

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

As titled.

Related issue number
--------------------

Fixes #1234

### CODE REVIEW

File: k8s/config/manager/kustomization.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: vineyardcloudnative/vineyard-operator
+  newName: localhost:5001/vineyard-operator
```
Comment (sighingnow) [inline k8s/config/manager/kustomization.yaml:7]: DON'T use `localhost:5001` as the default value.

File: k8s/pkg/templates/recover/job.yaml [hunk 2]
```diff
@@ -55,13 +55,13 @@ spec:
...
           name: recover-path
       volumes:
       - name: vineyard-sock
         hostPath:
           # The path should be changed to the path provided by users.
-          path: {{ $config.VineyardSockPath }}
+          path: {{ $config.Config.VineyardSockPath }}
       - name: recover-path
         persistentVolumeClaim:
           claimName: {{ $config.BackupPVCName }}
```
Comment (sighingnow) [inline k8s/pkg/templates/recover/job.yaml:67]: Is the nested `config.Config` avoidable?

**Review by sighingnow — COMMENTED** [review]
Remove the binary `vineyardctl` from the commit.

File: k8s/cmd/commands/util/certmanager.go [hunk 3]
```diff
@@ -45,3 +57,20 @@ func GetCertManagerManifests(certManagerManifestURL string) (Manifests, error) {
...
+	if flags.CertManagerVersion == flags.DefaultCertManagerVersion {
+		certManagerManifests, err = getCertManagerManifestsFromLocal()
+		if err != nil {
+			log.Fatal(err, "failed to get cert-manager manifests from local")
+		}
+	} else {
+		certManagerManifests, err = getCertManagerManifestsFromURL(getCertManagerURL())
+		if err != nil {
+			log.Fatal(err, "failed to get cert-manager manifests from remote")
+		}
```
Comment (sighingnow) [inline k8s/cmd/commands/util/certmanager.go:73]: Use the bundled cert-manager directly. I doesn't make sense to use a user-specific, _untested_ version of cert-manager for vineyard-operator, to simplify the process.

You can just remove the check, and remember to update the corresponding documentation and command line arguments as well.

Comment (dashanji) [reply k8s/cmd/commands/util/certmanager.go:73]: Make sense to me.

File: k8s/cmd/commands/util/parse.go [hunk 4]
```diff
@@ -142,3 +112,55 @@ func ParsePVCSpec(pvcspec string) (*corev1.PersistentVolumeClaimSpec, error) {
...
+	if err != nil {
+		return nil, nil, err
+	}
+	return pvSpec, pvcSpec, nil
+}
+
+// SetPVAndPVC parse the pVAndPVC string to corev1.PersistentVolumeSpec
+// and corev1.PersistentVolumeClaimSpec, then set the relevant fields.
+func SetPVAndPVC(pVAndPVC string, pv *corev1.PersistentVolumeSpec,
+	pvc *corev1.PersistentVolumeClaimSpec) error {
```
Comment (sighingnow) [inline k8s/cmd/commands/util/parse.go:151]: Move `pv, pvc` to return value to make it more go-style.

**Pull request merged**